supervisor/hardware/manager.py

```diff
@@ -103,8 +103,12 @@ def _import_devices(self) -> None:
         # Exctract all devices
         for device in self._udev.list_devices():
             # Skip devices without mapping
-            if not device.device_node or self.helper.hide_virtual_device(device):
-                continue
+            try:
+              if not device.device_node or self.helper.hide_virtual_device(device):
+                  continue
+            except:
```

Comment:
This should not be such a broad exception handler.

Use the exception logger to get more details about which exception is raised.
```
              _LOGGER.exception("This is the exception:")
```

I am guessing the exception happens in `hide_virtual_device`? If so, even better might be to call the `hide_virtual_device` helper only on certain conditions?

Reply:
Thanks, I will add a log to when the exception occurs.. 

*CAVEAT* I am not a python developer - I learned and dug in over a weekend to find out where the problem was. 

I make the same assumption that its the hide_virtual causing the problems, but since device.device_node cannot throw excptions, I opted not to split the node check and hide_virtual_device functions.

The exception is logged in the bug report. (below) 
``` 
024-05-17 19:33:01.831 CRITICAL (MainThread) [supervisor.core] Fatal error happening on load Task <coroutine object HardwareManager.load at 0x7f8dad5780>: 'utf-8' codec can't decode byte 0x9f in position 7: invalid start byte```

Reply:
You need catch the exactly exception and not everthing

Reply:
Hm, it sounds as if udev returns something which can't be interpreted as UTF-8 string. I wonder what D-Bus property that is, if there is maybe a bug in pyudev or something. :thinking:  Maybe related to some partition names? https://github.com/pyudev/pyudev/issues/490

Can you check the output of `blkid` on the shell if there are partition labels with non-ASCII characters or something?

I really would like to see the full stack trace to better understand what is wrong here. The current code should print the full stack. Can you run the current code and post the log output with the stack trace?

Reply:
From blkid - didnt spot anything wrong there
```
khadas@debian12:~$ sudo blkid
/dev/mmcblk0p1: LABEL="BOOT" UUID="83af10f3-33c4-4bff-adc6-c4ed966ecdd4" BLOCK_SIZE="4096" TYPE="ext4" PARTUUID="01c20000-01"
/dev/mmcblk0p2: LABEL="ROOTFS" UUID="32bf2441-7557-446a-957d-df58dd29f35e" BLOCK_SIZE="4096" TYPE="ext4" PARTUUID="01c20000-02"
/dev/zram3: UUID="691092b9-e03d-462d-bf91-a05792b7b447" TYPE="swap"
/dev/zram1: UUID="0b47f70c-e4ef-4ecd-8dee-4f5b6f543a75" TYPE="swap"
/dev/zram4: UUID="85cd0967-2e13-428c-81b9-4904523ba8ee" TYPE="swap"
/dev/zram2: UUID="119c6e18-d0bc-4755-a406-37fa7d68c742" TYPE="swap"
```
I printed the exception and the node its barfing at..

```2024-05-22 12:09:38.663 INFO (MainThread) [supervisor.api] Starting API on 172.30.32.2
2024-05-22 12:09:38.879 ERROR (MainThread) [supervisor.hardware.manager] EXCEPTION : Device('/sys/devices/virtual/misc/hifi4dsp0')
Traceback (most recent call last):
  File "/usr/src/supervisor/supervisor/hardware/manager.py", line 107, in _import_devices
    if not device.device_node or self.helper.hide_virtual_device(device):
           ^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/pyudev/device/_device.py", line 709, in device_node
    return ensure_unicode_string(node) if node is not None else None
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/pyudev/_util.py", line 56, in ensure_unicode_string
    value = value.decode(sys.getfilesystemencoding())
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
UnicodeDecodeError: 'utf-8' codec can't decode byte 0x9f in position 7: invalid start byte
2024-05-22 12:09:38.923 INFO (MainThread) [supervisor.hardware.monitor] Started Supervisor hardware monitor
2024-05-22 12:09:38.928 INFO (MainThread) [supervisor.dbus.manager] Connected to system D-Bus.
```

Reply:
Uh I see, I guess it is that `(null)` there, so even `udevadm` struggles?

I wonder if there is really a weird character in the driver somewhere that such a weird modalias appears? :thinking:

Reply:
IMHO, if this is a valid character as a MODALIAS, then maybe this error should be handled in pyudev.

Reply:
It should be pyudev ... but their forum has bug reports about the UTF-8 problems. Doesnt look like much activity on resolving it. 

Although, taking raw data that is not sanity checked is a recipe for a kicking! :D